### ISSUE TITLE
Compile Time error :: `max(int64, int64)` returns a `number` not a `int64`

### ISSUE THREAD

**Issue by tinkerer-red** [YoYoGames/GameMaker-Bugs#8021]
### Description

I would imagine that many of the other compile time ready math expressions would suffer from a similar issue.

### Steps To Reproduce

```gml
var _a1 = 0x1122334455667788;
var _a2 = 0x8877665544332211;
var _a3 = 0x7FFFFFFFFFFFFFFF;
var _a4 = 0x5566778811223344;
show_debug_message(typeof(max(_a1, _a2, _a3, _a4)))
show_debug_message(typeof(max(0x1122334455667788, 0x8877665544332211, 0x7FFFFFFFFFFFFFFF, 0x5566778811223344)))
```

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.1 (Monthly), 2022.0.2 (LTS)

### Which operating system(s) are you seeing the problem on?

Windows 10

### Are you running GameMaker from inside your Steam library?

None

### Sample Package Attached?

- [X] I have attached my bug package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by rwkay** [comment]
Duplicate of #8022

**Comment by tinkerer-red** [comment]
@rwkay The issue for this is a type difference, the issue for 8022 is an incorrect value.
Just to be clear, by duplicate I assume you mean the same fix, fixed both? and there wasn't miscommunication in the bug report? As I would imagine a quick glance both reports look identical.

**Comment by rwkay** [comment]
It was the same fix - basically a mixed array (doubles and 64 bit integers) cannot be optimised by the compiler - the same fix for #8022 fixed this one.

**Comment by scott-dunbar** [comment]
Verified fixedin  IDE v2024.1100.0.653  Runtime v2024.1100.0.679
